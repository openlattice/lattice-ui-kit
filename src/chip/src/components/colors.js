import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
