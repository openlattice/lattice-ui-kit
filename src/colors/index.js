/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
