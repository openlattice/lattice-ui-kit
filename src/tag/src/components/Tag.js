// @flow
import styled from 'styled-components';

import * as Colors from '../../../colors';
import { getStyleVariation } from '../../../utils/StyleUtils';

const {
  GREEN_1,
  NEUTRAL,
  PURPLE,
  RED_1,
  WHITE,
  YELLOW_1,
} = Colors;

const getBackgroundColor = getStyleVariation('mode', {
  danger: RED_1,
  neutral: NEUTRAL.N700,
  primary: PURPLE.P300,
  secondary: PURPLE.P100,
  success: GREEN_1,
  warning: YELLOW_1
}, NEUTRAL.N100);

const getFontColor = getStyleVariation('mode', {
  danger: WHITE,
  neutral: WHITE,
  primary: WHITE,
  secondary: PURPLE.P300,
  success: WHITE,
  warning: NEUTRAL.N700
}, NEUTRAL.N700);

const Tag = styled.span`
  border-radius: 3px;
  font-size: 12px;
  font-weight: 700;
  margin: 0 5px;
  padding: 0 5px;
  text-transform: uppercase;

  background-color: ${getBackgroundColor};
  color: ${getFontColor};
`;

export default Tag;
