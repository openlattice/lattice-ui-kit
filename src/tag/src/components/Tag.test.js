import React from 'react';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Tag from './Tag';

import * as Colors from '../../../colors';

const {
  GREEN,
  NEUTRAL,
  PURPLE,
  RED,
  YELLOW,
} = Colors;

describe('Tag', () => {
  describe('snapshots', () => {

    test('should match snapshot', () => {
      const wrapper = mount(<Tag>tag</Tag>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

  describe('modes', () => {

    test('default', () => {
      const wrapper = mount(<Tag>tag</Tag>);
      expect(wrapper.find(Tag)).toHaveStyleRule('color', NEUTRAL.N700);
      expect(wrapper.find(Tag)).toHaveStyleRule('background-color', NEUTRAL.N100);
    });

    test('danger', () => {
      const wrapper = mount(<Tag mode="danger">tag</Tag>);
      expect(wrapper.find(Tag)).toHaveStyleRule('color', 'white');
      expect(wrapper.find(Tag)).toHaveStyleRule('background-color', RED.R300);
    });

    test('neutral', () => {
      const wrapper = mount(<Tag mode="neutral">tag</Tag>);
      expect(wrapper.find(Tag)).toHaveStyleRule('color', 'white');
      expect(wrapper.find(Tag)).toHaveStyleRule('background-color', NEUTRAL.N700);
    });

    test('primary', () => {
      const wrapper = mount(<Tag mode="primary">tag</Tag>);
      expect(wrapper.find(Tag)).toHaveStyleRule('color', 'white');
      expect(wrapper.find(Tag)).toHaveStyleRule('background-color', PURPLE.P300);
    });

    test('secondary', () => {
      const wrapper = mount(<Tag mode="secondary">tag</Tag>);
      expect(wrapper.find(Tag)).toHaveStyleRule('color', PURPLE.P300);
      expect(wrapper.find(Tag)).toHaveStyleRule('background-color', PURPLE.P100);
    });

    test('success', () => {
      const wrapper = mount(<Tag mode="success">tag</Tag>);
      expect(wrapper.find(Tag)).toHaveStyleRule('color', 'white');
      expect(wrapper.find(Tag)).toHaveStyleRule('background-color', GREEN.G300);
    });

    test('warning', () => {
      const wrapper = mount(<Tag mode="warning">tag</Tag>);
      expect(wrapper.find(Tag)).toHaveStyleRule('color', NEUTRAL.N700);
      expect(wrapper.find(Tag)).toHaveStyleRule('background-color', YELLOW.Y300);
    });

  });
});
