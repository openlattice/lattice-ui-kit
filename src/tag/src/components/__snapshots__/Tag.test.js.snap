// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag snapshots should match snapshot 1`] = `
.c0 {
  border-radius: 3px;
  font-size: 12px;
  font-weight: 700;
  margin: 0 5px;
  padding: 0 5px;
  text-transform: uppercase;
  background-color: #DFDFE3;
  color: #56565A;
}

<Tag>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Tag-sc-1sp059-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "border-radius:3px;font-size:12px;font-weight:700;margin:0 5px;padding:0 5px;text-transform:uppercase;background-color:",
            [Function],
            ";color:",
            [Function],
            ";",
          ],
        },
        "displayName": "Tag",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Tag-sc-1sp059-0",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <span
      className="c0"
    >
      tag
    </span>
  </StyledComponent>
</Tag>
`;
